userspace/falco/app_actions/load_plugins.cpp

```diff
@@ -27,121 +27,34 @@ application::run_result application::load_plugins()
-			// overhead of additional mappings at runtime, but we may consider
-			// mapping the two indexes under something like std::unordered_map in the future.
-			bool added = false;
-			auto source_idx = m_state->inspector->get_plugin_manager()->source_idx_by_plugin_id(plugin->id(), added);
-			auto source_idx_engine = m_state->engine->add_source(plugin->event_source(), filter_factory, formatter_factory);
-			if (!added || source_idx != source_idx_engine)
-			{
-				return run_result::fatal("Could not add event source in the engine: " + plugin->event_source());
-			}
-		}
-	}
+	// Initialize the offline inspector. This is used to load all the configured
+	// plugins in order to have them available every time we need to access
+	// their static info. If Falco is in capture mode, this inspector is also
+	// used to open and read the trace file
+	m_state->offline_inspector.reset(new sinsp());
 
-	// Iterate over the plugins with extractor capability and add them to the
-	// filtercheck list of their compatible sources
-	std::vector<const filter_check_info*> filtercheck_info;
-	for(const auto& p : m_state->inspector->get_plugin_manager()->plugins())
+	// Load all the configured plugins
+	for(auto &p : m_state->config->m_plugins)
 	{
-		if (!(p->caps() & CAP_EXTRACTION))
-		{
-			continue;
-		}
-
-		bool used = false;
-		for (auto &it : m_state->plugin_filter_checks)
-		{
-			// check if the event source is compatible with this plugin
-			if (p->is_source_compatible(it.first))
-			{
-				// check if some fields are overlapping on this event sources
-				filtercheck_info.clear();
-				it.second.get_all_fields(filtercheck_info);
-				for (auto &info : filtercheck_info)
-				{
-					for (int32_t i = 0; i < info->m_nfields; i++)
-					{
-						// check if one of the fields extractable by the plugin
-						// is already provided by another filtercheck for this source
-						std::string fname = info->m_fields[i].m_name;
-						for (auto &f : p->fields())
-						{
-							if (std::string(f.m_name) == fname)
-							{
-								return run_result::fatal(
-									"Plugin '" + p->name()
-									+ "' supports extraction of field '" + fname
-									+ "' that is overlapping for source '" + it.first + "'");
-							}
-						}
-					}
-				}
-
-				// add plugin filterchecks to the event source
-				it.second.add_filter_check(sinsp_plugin::new_filtercheck(p));
-				used = true;
-			}
-		}
-		if (!used)
+		falco_logger::log(LOG_INFO, "Loading plugin '" + p.m_name + "' from file " + p.m_library_path + "\n");
+		auto plugin = m_state->offline_inspector->register_plugin(p.m_library_path);
+		m_state->plugin_configs.insert(p, plugin->name());
+		if(plugin->caps() & CAP_SOURCING)
 		{
-			return run_result::fatal("Plugin '" + p->name()
-				+ "' has field extraction capability but is not compatible with any enabled event source");
+			auto sname = plugin->event_source();
+			m_state->sources.insert(empty_src, sname);
+			m_state->loaded_sources.insert(sname);
```

Comment:
So, what's the difference between `sources` and `loaded_sources` ?

Reply:
See comment above. I agree that `sources` is a poor naming decision. Do you have any suggestions? Maybe we should go for something simple like `source_infos` (which is what this really is about)?

Reply:
Yeah i agree, `sources_info` is a much better name imho :)